external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -539,3 +569,77 @@ class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
 std::unique_ptr<Pass> mlir::tosa::createTosaPartitionPass() {
   return std::make_unique<TosaPartitionPass>();
 }
+
+std::unique_ptr<Pass>
+mlir::tosa::createTosaPartitionPass(llvm::function_ref<bool(Operation *)> &pred,
+                                    std::string attrName, bool noFront) {
```

Comment:
use StringRef

Reply:
Also should have a test for predecessors and successors. I suggest a class in the header that has overrideable methods to test each.